src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitationMethod.cpp

```diff
@@ -0,0 +1,259 @@
+    transformation_model_params_ = transformation_model_params;
+  }
+
+  void AbsoluteQuantitationMethod::getTransformationModel(String& transformation_model, Param& transformation_model_params)
+  {
+    transformation_model = transformation_model_;
+    transformation_model_params = transformation_model_params_;
+  }
+  
+  void AbsoluteQuantitationMethod::setActualConcentration(const double& actual_concentration)
+  {
+    actual_concentration_ = actual_concentration;
+  }
+
+  void AbsoluteQuantitationMethod::getActualConcentration(double& actual_concentration)
+  {
+    actual_concentration = actual_concentration_;
+  }
+  
+  void AbsoluteQuantitationMethod::setStatistics(const int& n_points, const double& correlation_coefficient)
+  {
+    n_points_ = n_points;
+    correlation_coefficient_ = correlation_coefficient;
+  }
+  
+  void AbsoluteQuantitationMethod::getStatistics(int& n_points, double& correlation_coefficient)
+  {
+    n_points = n_points_;
+    correlation_coefficient = correlation_coefficient_;
+  }
+
+  Param AbsoluteQuantitationMethod::fitTransformationModel(const String & transformation_model,
+    const TransformationModel::DataPoints& data,
+    const Param& transformation_model_params)
+  {
+    Param params;
+    if (transformation_model == "TransformationModelLinear")
+    {
+      TransformationModelLinear tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    else if (transformation_model == "TransformationModelBSpline")
+    {
+      TransformationModelBSpline tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    else if (transformation_model == "TransformationModelInterpolated")
+    {
+      TransformationModelInterpolated tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    else if (transformation_model == "TransformationModelLowess")
+    {
+      TransformationModelLowess tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    else
+    {
+      LOG_INFO << "TransformationModel " << transformation_model << " is not supported.";
+      LOG_INFO << "default TransformationModel will be used.";
+      TransformationModel tm(data, transformation_model_params);
+      params = tm.getParameters();
+    }
+    return params;
+  }
+  
+  double AbsoluteQuantitationMethod::evaluateTransformationModel(const String & transformation_model,
+    const double& datum,
+    const Param& transformation_model_params)
+  {
+    double result = datum;
+    TransformationModel::DataPoints data;
+    if (transformation_model == "TransformationModelLinear")
+    {
+      TransformationModelLinear tm(data, transformation_model_params);
+      tm.invert();
+      result = tm.evaluate(datum);
+    }
+    else if (transformation_model == "TransformationModelBSpline")
+    {
+      TransformationModelBSpline tm(data, transformation_model_params);
+      // tm.invert(); // not supported
```

Comment:
I dont think I follow the logic here, how can the result be correct if it requires inversion of the model but you only invert for the linear case?

Reply:
At the moment I don't believe that there is a way to invert the other models, unless I am mistaken.

Reply:
well, the easy way would be to switch  x/y coordinates and then recompute, right? or am I missing something?

Reply:
I believe you would have to invert the model.  This is what the software vendors do as well.